src/librustc_trait_selection/traits/wf.rs

```diff
@@ -356,9 +375,9 @@ impl<'a, 'tcx> WfPredicates<'a, 'tcx> {
         }
     }
 
-    /// Pushes the obligations required for an array length to be WF
+    /// Pushes the obligations required for a constant value to be WF
     /// into `self.out`.
-    fn compute_array_len(&mut self, constant: ty::Const<'tcx>) {
+    fn compute_const(&mut self, constant: ty::Const<'tcx>) {
         if let ty::ConstKind::Unevaluated(def_id, substs, promoted) = constant.val {
```

Comment:
This should be an exhaustive match and you'll want to handle `ConstKind::Infer` a bit like this: https://github.com/rust-lang/rust/blob/f4c675c476c18b1a11041193f2f59d695b126bc8/src/librustc_trait_selection/traits/wf.rs#L569-L593

(You can probably want to remove the `return bool` and the `assert!` around the recursive call)